engine/swarm/secrets.md

```diff
@@ -0,0 +1,857 @@
+    WordPress user, which is completely different from the password WordPress
+    uses to access MySQL. Store this password securely, such as in a password
+    manager. You will need it to log into WordPress after
+    [rotating the secret](#example-rotate-a-secret).
+
+    Go ahead and write a blog post or two or install some WordPress plugins to
+    verify that WordPress is fully operational.
+
+8.  Do not clean up any services or secrets if you intend to proceed to the next
+    example, which demonstrates how to rotate the MySQL root password.
+
+
+### Example: Rotate a secret
+
+This example builds upon the previous one. In this scenario, you create a new
+secret with a new MySQL password, update the `mysql` and `wordpress` services to
+use it, then remove the old secret.
+
+**Note**: Changing the password on a MySQL database involves running extra
+queries or commands, as opposed to just changing a single environment variable
+or a file, since the image only sets the MySQL password if the database doesn’t
+already exist, and MySQL stores the password within a MySQL database by default.
+Rotating passwords or other secrets will often involve additional steps outside
+of Docker.
+
+1.  Create the new password and store it in file in the `tempdir` directory
+    (which is bind-mounted into the MySQL container) and also as a secret named
+    `mysql_password_v2`.
+
+    ```bash
+    $ openssl rand -base64 20 | docker secret create mysql_password_v2
+
+    z05lwief4u8wk6jc89xx2l4es
+    ```
+
+2.  Update the MySQL service to give it access to both the old and new secrets.
+    Remember that you cannot update or rename a secret, but you can revoke a
+    secret and grant access to it using a new target filename.
+
+    ```bash
+    $ docker service update \
+                     --secret-rm mysql_password mysql
+
+    $ docker service update \
+                     --secret-add source=mysql_password,target=old_mysql_password \
+                     --secret-add source=mysql_password_v2,target=mysql_password \
+                     mysql
+    ```
+
+    Updating a service causes it to restart, and when the MySQL service restarts
+    the second time, it has access to the old secret under
+    `/run/secrets/old_mysql_password` and the new secret under
+    `/run/secrets/mysql_password`.
+
+    Even though the MySQL service has access to both the old and new secrets
+    now, the MySQL root password has not yet been changed.
+
+3.  Now, change the MySQL password for the `wordpress` user using the `mysql`
+    CLI. This command reads the old and new password from the files in
+    `/run/secrets` but does not expose them on the command line or save them in
+    the shell history.
+
+    Do this quickly and move on to the next step, because WordPress will lose
+    the ability to connect to MySQL.
+
+    First, find the ID of the `mysql` container task.
+
+    ```bash
+    $ docker ps | grep mysql |awk {'print $1'}
```

Comment:
It's better to use the built-in filter functionality here (and `-q` to only show the `ID`;

```
docker ps -q --filter name=mysql
```